src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariantContext.java

```diff
@@ -0,0 +1,280 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Variant context with additional method to mind the structural variant specific information from structural
+ * variant records.
+ */
+public class StructuralVariantContext extends VariantContext {
+
+    private static final long serialVersionUID = 1L;
+
+    private int end = -1;
+    private int length = -1;
+
+    /**
+     * Returns a instance given a plain {@link VariantContext} instance.
+     * <p>
+     *     This method will fail with a {@link IllegalArgumentException} if the input variant context is not
+     *     structural. Currently we only check that it has been properly annotated with the {@value VCFConstants#SVTYPE}
+     *     annotation.
+     * </p>
+     * @param vc the input variant context.
+     * @throws IllegalArgumentException if {@code vc} is {@code null} or does not seem to be
+     * a structural variant context based on its annotations.
+     * @return never {@code null}, potentially the same as the input if it happens to be an instance of this class.
+     */
+    public static StructuralVariantContext create(final VariantContext vc) {
+        if (vc instanceof StructuralVariantContext) {
+            return (StructuralVariantContext) vc;
+        } else {
+            checkIsStructuralVariantContext(vc);
+            return new StructuralVariantContext(vc);
+        }
+    }
+
+    private static void checkIsStructuralVariantContext(final VariantContext vc) {
+        Utils.nonNull(vc, "the input variant context must not be null");
+        Utils.nonNull(vc.getAttributeAsString(VCFConstants.SVTYPE, null), "the SVTYPE annotation is missing");
+    }
+
+    private StructuralVariantContext(final VariantContext other) {
+        super(other);
+    }
+
+    /**
+     * The end position of this variant context.
+     * This end position is determined with this priorityq:
+     * <ul>
+     *     <li>the value of the annotation {@value VCFConstants#END_KEY} if present, otherwise</li>
+     *     <li>the value that results from applying the length annotated under {@value  GATKSVVCFConstants#SVLEN} based on the structural type if such annoation is present, otherwise</li>
+     *     <li>the variant context start position + the length of the reference allele - 1.</li>
+     * </ul>
+     * @return whatever is returned by {@link #getStart()} or greater.
+     */
+    @Override
+    public int getEnd() {
+        if (end <= 0) {
+            if (hasAttribute(VCFConstants.END_KEY)) {
+                end = getAttributeAsInt(VCFConstants.END_KEY, -1);
+            } else if (getStructuralVariantType() == StructuralVariantType.DEL) {
+                end = getStart() + getStructuralVariantLength();
+            } else {
+                end = super.getEnd();
+            }
+        }
+        return end;
+    }
+
+    /**
+     * Returns the assembly ids for this context's structural variant.
```

Comment:
In addition, the name of this function is easily confused with `getContig()`. 
I suggest `getSupportingContigs()`